File: server/model/pet_todo/request/pet_todo.go
package request

import (
	"github.com/Lzzzzzzy/UPet/server/model/common"
)

type PetTodoCondition struct {
	Date  string `json:"date" form:"date"`   // 日期
	PetId uint   `json:"petId" form:"petId"` // 宠物id
}

type PetTodoInfo struct {
	Title      string             `json:"title" form:"title"`           // 标题
	Remark     string             `json:"remark" form:"remark"`         // 备注
	Remind     bool               `json:"remind" form:"remind"`         // 是否提醒， 0：否，1：是
	RemindDate []string           `json:"remindDate" form:"remindDate"` // 提醒日期
	RemindTime string             `json:"remindTime" form:"remindTime"` // 提醒时间
	Complete   bool               `json:"complete" form:"complete"`     // 是否完成，0：否，1：是
	Type       uint               `json:"type" form:"type"`             // 提醒类型， 0：日常记录， 1：待办事项
	Color      uint               `json:"color" form:"color" `          // 背景颜色
	TodoTime   *common.CustomTime `json:"todoTime" form:"todoTime" `    // 待办时间
	PetId      uint               `json:"petId" form:"petId" `          // 宠物ID
}

type PetTodoMarkCondition struct {
	MinDate string `json:"minDate" form:"minDate"` //最小日期
	MaxDate string `json:"maxDate" form:"maxDate"` // 最大日期
}

type PetTodoComplete struct {
	Complete bool `json:"complete" form:"complete"` // 是否完成
}

File: web/src/service/api/petTodo.ts
import { request } from '../request';
import { Pet } from "@/typings/pet";


interface petTodosReq {
  petId: number,
  date: string,
}

interface petTodo {
  id?: number,
  title: string,
  todoTime: string,
  remark: string,
  remind: boolean,
  remindDate: Array<string>,
  remindTime: string,
  complete: boolean,
  type: number,
  color: number,
}

/** 添加宠物待办 */
export async function addPetTodo(data: petTodo) {
  const resp = await request.post('/api/pet-todo', data, {
    useErrMsg: false
  });
  return resp.success;
}

/** 查询宠物待办 */
export async function getPetTodosOnPagenation(condition: petTodosReq) {
  const resp = await request.get<Array<Pet.PetTodo>>('/api/pet-todos', condition, {
    useErrMsg: false
  });
  return resp.success;
}

/** 查询宠物待办标记 */
export async function getPetTodosMark(dates: Pet.PetMarkParam) {
  const resp = await request.post<Array<Pet.PetTodo>>('/api/pet-todos/mark', dates, {
    useErrMsg: false
  });
  return resp.success;
}

/**
 * 更新宠物待办完成情况
 *
 * @param {number} todoId - The ID of the todo item to update.
 * @param {boolean} complete - The new complete status of the todo item.
 * @return {Promise} A promise that resolves to the success status of the update operation.
 */
export async function updateTodoCompleteStatus(todoId: number, complete: boolean) {
  const resp = await request.put(`/api/pet-todo/${todoId}/complete`, { complete }, {
    useErrMsg: false
  });
  return resp.success;
}

/**
 * 搜索宠物待办信息
 *
 * @param {string} content - Need search content.
 * @return {Promise} A promise that resolves to the success status of the update operation.
 */
export async function searchPetTodoInfo(content: string) {
  const resp = await request.get(`/api/pet-todos/infos`, { content }, {
    useErrMsg: false
  });
  return resp.success;
}

/** 编辑宠物待办 */
export async function editPetTodo(data: petTodo, id: number) {
  const resp = await request.put(`/api/pet-todo/${id}`, data, {
    useErrMsg: false
  });
  return resp.success;
}

/** 删除宠物待办 */
export async function deletePetTodo(id: number) {
  const resp = await request.delete(`/api/pet-todo/${id}`, {}, {
    useErrMsg: false
  });
  return resp.success;
}

/**
 * 根据id查询宠物待办信息
 *
 * @param {number} todoId - Need search todo id.
 * @return {Promise} A promise that resolves to the success status of the update operation.
 */
export async function getPetTodoInfoById(todoId: number) {
  const resp = await request.get(`/api/pet-todo/${todoId}`, null, {
    useErrMsg: false
  });
  return resp.success;
}


/**
 * 删除宠物待办信息
 *
 * @param {number} todoId - Need search todo id.
 * @return {Promise} A promise that resolves to the success status of the update operation.
 */
export async function deletePetTodoInfoById(todoId: number) {
  const resp = await request.delete(`/api/pet-todo/${todoId}`, null, {
    useErrMsg: false
  });
  return resp.success;
}
